clang/test/Driver/sycl-bitfield-accesses.cpp

```diff
@@ -0,0 +1,44 @@
+// RUN: %clangxx -O0 -fsycl -fsycl-targets=spir64 %s -S -emit-llvm -o- | FileCheck -check-prefix=CHECK-FINE %s
+// RUN: %clangxx -O0 -fsycl %s -S -emit-llvm -o- | FileCheck -check-prefix=CHECK-COARSE %s
```

Comment:
spir64 is a default target, so these command are doing the same thing.
It passes because with `-fsycl` compiler emits llvm for both host and spir64 target, so both checks can be found in both outputs.
I suggest:
1. Leave default optimization level.
2. Replace -fsycl with -fsycl-device-only.
3. Remove second run command.

Reply:
Are we really ok with device/host lowering the same type differently? What if I capture it as a kernel arg? Or pass in memory...

Reply:
We need the O0 optimization level. Otherwise the variables get optimized out. I made the other changes.

Thanks

Reply:
> We need the O0 optimization level. Otherwise the variables get optimized out. I made the other changes.

I think we can drop O0 with test changes. See https://github.com/intel/llvm/pull/7989/files#r1073042087.

Reply:
> Are we really ok with device/host lowering the same type differently? What if I capture it as a kernel arg? Or pass in memory...

That might be a valid concern. @asudarsa, could you add more test cases when bitfield struct contains types < 8 bits? The bitfield must be "device copiable" to qualify for a kernel parameter type.
If `-ffine-grained-bitfield-accesses` changes the size of the bitfield, we must set the same value for both host and device to make sure that bitfield object is "device copiable".

Reply:
Probably this comment addresses < 8 bits concern https://github.com/intel/llvm/pull/7989#pullrequestreview-1245119412

Reply:
Hi @bader 

Does the comment pointed to by @MrSidims address your concern? I can add more test cases if needed. How can we check if an object is device copiable?

Thanks